src/app/(app)/course/[courseId]/project-sets/[projectSetId]/(hooks)/useProjectSearch.tsx

```diff
@@ -0,0 +1,48 @@
+'use client'
+
+import {createContext, useContext, useEffect, useState, type PropsWithChildren, type FC} from "react"
+import {useProjectsContext} from "@/app/(app)/course/[courseId]/project-sets/[projectSetId]/(hooks)/useProjects"
+
+type ProjectSearchContextType = {
+  searchText: string
+  setSearchText: (searchText: string) => void
+}
+
+const ProjectSearchContext = createContext<ProjectSearchContextType>({
+  searchText: '',
+  setSearchText: () => {},
+})
+
+const useProjectSearch = () => {
+  const {projects: allProjects, setDisplayProjects, currentProject, setCurrentProject} = useProjectsContext()
+  const [searchText, setSearchText] = useState('')
+
+  useEffect(() => {
+    const filteredProjects = allProjects.filter(project => project.name.toLowerCase().includes(searchText.toLowerCase()))
+    setDisplayProjects(filteredProjects)
+    setSearchText(searchText)
+
+    // If the current project is not in the filtered projects, set the current project to the first filtered project
```

Comment:
idea;

I think a nice UX flow would be to have the right side of the page render a nice empty state whenever the search changes. Then make it possible to set null for the current project. Then whenever the search changes we set the current project to null.